src/app/components/case/Workflow/Workflow.tsx

```diff
@@ -23,11 +24,16 @@ const Div = styled.div`
 const Workflow: React.FC<Props> = ({ id }) => {
   const [, { execPost }] = useTaskComplete({ lazy: true })
   const [data, { isBusy, execGet }] = useCaseWorkflows(id)
+  const [caseData] = useCase(id)
 
   const workflows = data?.results ?? []
   const columns = getColumns(execPost)
+  const isClosed = caseData?.end_date !== null
 
-  if (isBusy) {
+  const shouldPoll = !isClosed || workflows.length > 0
```

Comment:
The updated polling behavior:

- Always poll when case is open (regardless of workflow amount)
- Poll when case is closed but has workflows (e.g. potentially has stale workflow data after closing a case)
- Don't poll when case is closed and has no workflows

Polling is stopped when data is received, so in practice the `/api/v1/cases/{id}/workflows/` should only be fetched once.